client/src/components/atoms/NotificationListItem.vue

```diff
@@ -58,17 +66,35 @@ const props = defineProps({
   },
 })
 
+const { mutate: markNotificationRead } = useMutation(MARK_NOTIFICATION_READ, {
+  refetchQueries: ["currentUserNotifications"],
+})
+
+const handleClick = async (notification_id) => {
+  await markNotificationRead({
+    notification_id: notification_id,
+  })
+}
+
 /**
  * The parsed translation key to use for the supplied note
  */
 const tKey = computed(() => {
-  return `notifications.${props.note.type}.short`
+  return `notifications.${props.note.data.type}.short`
 })
 /**
  * Flattened version of the note for passing to i18n
  */
 const flattened = computed(() => {
-  return flatten(props.note)
+  const flattened = flatten(props.note)
+
+  const mutated_keys = {}
+  Object.keys(flattened).forEach((k) => {
+    // console.log("k", k.replaceAll("data.", "").replaceAll(".", "_"))
+    mutated_keys[k.replaceAll("data.", "").replaceAll(".", "_")] = flattened[k]
+  })
+  console.log("mutated", mutated_keys)
+  return mutated_keys
```

Comment:
```suggestion
  return flatten(props.note, {delimiter: "_")
```

Reply:
That delimiter parameter is nice, however, the problem is that the flattened data is returned with `data_` prepended to every property name. So, `data_` has to be removed in order for the mapping to work. The alternative is to update the translation files by nesting them inside a `data` object and I don't know if that's better.

Reply:
After trying both, I've decided it's better, so this has been refactored.

Reply:
I mean you could flatten(props.note.data, ....  no?

Reply:
Then other top-level properties like `id` won't be included.